yt_dlp/utils.py

```diff
@@ -410,54 +410,126 @@ def xpath_attr(node, xpath, key, name=None, fatal=False, default=NO_DEFAULT):
     return n.attrib[key]
 
 
-def get_element_by_id(id, html):
-    """Return the content of the tag with the specified ID in the passed HTML document"""
-    return get_element_by_attribute('id', id, html)
+def get_element_by_id(id, html, return_content=True):
+    """Return the content or the tag with the specified ID in the passed HTML document"""
+    return get_element_by_attribute('id', id, html, return_content=return_content)
 
 
-def get_element_by_class(class_name, html):
-    """Return the content of the first tag with the specified class in the passed HTML document"""
-    retval = get_elements_by_class(class_name, html)
+def get_element_by_class(class_name, html, return_content=True):
+    """Return the content or the first tag with the specified class in the passed HTML document"""
+    retval = get_elements_by_class(class_name, html, return_content)
     return retval[0] if retval else None
 
 
-def get_element_by_attribute(attribute, value, html, escape_value=True):
-    retval = get_elements_by_attribute(attribute, value, html, escape_value)
+def get_element_by_attribute(attribute, value, html, escape_value=True, return_content=True):
+    retval = get_elements_by_attribute(attribute, value, html, escape_value, return_content)
     return retval[0] if retval else None
 
 
-def get_elements_by_class(class_name, html):
-    """Return the content of all tags with the specified class in the passed HTML document as a list"""
+def get_elements_by_class(class_name, html, return_content=True):
+    """Return the content or all tags with the specified class in the passed HTML document as a list"""
     return get_elements_by_attribute(
         'class', r'[^\'"]*\b%s\b[^\'"]*' % re.escape(class_name),
-        html, escape_value=False)
+        html, escape_value=False, return_content=return_content)
 
 
-def get_elements_by_attribute(attribute, value, html, escape_value=True):
-    """Return the content of the tag with the specified attribute in the passed HTML document"""
+def get_elements_by_attribute(attribute, value, html, escape_value=True, return_content=True):
+    """Return the content or the tag with the specified attribute in the passed HTML document"""
```

Comment:
returning values by an argument is generally bad practice. Better to have separate functions for it. Maybe `get_element_html_by_class` etc. You can have a `get_elements_text_and_html_by_attribute` that returns `text, whole` over which everything else can be wrapped